up_client_socket/rust/utransport-socket/src/lib.rs

```diff
@@ -0,0 +1,520 @@
+             listener_map,
+         })
+     }
+ 
+     /// # Panics
+     ///
+     /// Will panic if failed to parse message
+      fn socket_init(&mut self) -> Result<(), UStatus>{
+        let socket_clone = match self.socket_sync.try_clone() {
+            Ok(socket) => socket,
+            Err(err) => {
+          
+                error!("Issue in cloning sync socket: {}", err);
+                return Err(UStatus::fail_with_code(UCode::INTERNAL, "Issue in cloning socket_sync"));
+            }
+        };
+    
+    let listener_map_clone = self.listener_map.clone();
+    
+    let mut self_copy = Self {
+        socket_sync: socket_clone,
+        listener_map: listener_map_clone,
+    };
+    task::spawn(async move { self_copy.dispatcher_listener().await });
+   
+    
+    Ok(())
+}
+    async  fn dispatcher_listener(&mut self) {
+        debug!("started listener for dispatcher");
+         loop {
+             // Receive data from the socket
+             let mut buffer: [u8; BYTES_MSG_LENGTH] = [0; BYTES_MSG_LENGTH];
+             let bytes_read = match self.socket_sync.read(&mut buffer) {
+                 Ok(bytes_read) => bytes_read,
+                 Err(e) => {
+                     debug!("Socket error: {}", e);
+                     break;
+                 }
+             };
+ 
+             // Check if no data is received
+             if bytes_read == 0 {
+                 continue;
+             }
+ 
+             let umessage_result = UMessage::parse_from_bytes(&buffer[..bytes_read]);
+             let umessage = match umessage_result {
+                 Ok(umsg) => umsg,
+                 Err(err) => {
+                     error!("Failed to parse message: {}", err);
+                     continue;
+                 }
+                };
+             
+             match umessage
+                 .attributes
+                 .type_
+                 .enum_value_or(UMessageType::UMESSAGE_TYPE_UNSPECIFIED)
+             {
+                 UMessageType::UMESSAGE_TYPE_PUBLISH => {
+                     debug!("calling handle publish....");
+                     match self.check_all_listeners(&umessage.attributes.source.clone(), umessage){
+                        Ok(_) => {
+                            // The function call succeeded
+                            debug!("All listeners checked successfully");
+                        }
+                        Err(err) => {
+                            // The function call failed with an error
+                            error!("Error checking listeners: {}", err);
+                           continue;
+                        }
+                    }
+                 }
+                 UMessageType::UMESSAGE_TYPE_NOTIFICATION | UMessageType::UMESSAGE_TYPE_UNSPECIFIED | UMessageType::UMESSAGE_TYPE_RESPONSE => (),
+                 UMessageType::UMESSAGE_TYPE_REQUEST => {
+                    match self.check_all_listeners(&umessage.attributes.sink.clone(), umessage){
```

Comment:
Try for same idea as suggested [here](https://github.com/eclipse-uprotocol/up-tck/pull/23/files#r1597260805).